### PR TITLE
TS-4978: illegal memory access with ticket_key.filename

### PR DESCRIPTION
**Pull request by persiaAziz**
Avoiding illegal memory access by checking the return value of readConfigStringAlloc .

### CODE REVIEW

File: iocore/net/SSLConfig.cc [hunk 1]
```diff
@@ -253,8 +253,8 @@ SSLConfigParams::initialize()
   ats_free(CACertRelativePath);
 
 #if HAVE_OPENSSL_SESSION_TICKETS
-  REC_ReadConfigStringAlloc(ticket_key_filename, "proxy.config.ssl.server.ticket_key.filename");
```
Comment (SolidWallOfCode) [inline iocore/net/SSLConfig.cc]: Why not just embed the call in the `if`?
```
if (REC_ERR_OKAY == REC_ReadConfigString(...)) {
```

File: iocore/net/SSLConfig.cc [hunk 2]
```diff
@@ -112,7 +112,7 @@ SSLConfigParams::cleanup()
   ats_free_null(client_cipherSuite);
   ats_free_null(dhparamsFile);
   ats_free_null(ssl_wire_trace_ip);
-  ats_free_null(ticket_key_filename);
+  ticket_key_filename = (char *)ats_free_null(ticket_key_filename);
```
Comment (SolidWallOfCode) [inline iocore/net/SSLConfig.cc]: Let's go bigger. Pull the code out of the constructor, make it a method, and then call that at the end of this method. Reset everything, really clean up.

**Comment by persiaAziz** [comment]
Please review @SolidWallOfCode

**Comment by SolidWallOfCode** [comment]
Looks good.

**Comment by zwoop** [comment]
Looks like we have merge conflicts now.

**Comment by persiaAziz** [comment]
I will do a local merge and then make a new  PR

**Comment by persiaAziz** [comment]
The conflict is resolved. Please check

**Pull request merged**